src/test/java/com/alvarium/annotators/SourceCodeAnnotatorTest.java

```diff
@@ -92,9 +92,9 @@ public void executeShouldReturnAnnotation() throws AnnotatorException, IOExcepti
                 final File checksumFile = dir.newFile("checksum");
     
                 final HashProvider hash = new HashProviderFactory().getProvider(config.getHash().getType());
-                String hashAndPath = hash.derive(Files.readAllBytes(f1.toPath())) + "  " + f1.toPath().toString() + "\n";
-                hashAndPath = hashAndPath + hash.derive(Files.readAllBytes(f2.toPath())) + "  " + f2.toPath().toString() + "\n";
-                final String checksum = hash.derive(hashAndPath.getBytes());
+                String hashAndPath = hash.derive(Files.readAllBytes(f1.toPath())).toLowerCase() + "  " + f1.toPath().toString() + "\n";
+                hashAndPath = hashAndPath + hash.derive(Files.readAllBytes(f2.toPath())).toLowerCase() + "  " + f2.toPath().toString() + "\n";
+                final String checksum = hash.derive(hashAndPath.getBytes()).toLowerCase();
```

Comment:
I have reviewed the recent fix made to the source-code annotator logic, it looks good, I tried it and it produces satisfied annotations, but I noticed that the corresponding test is failing. It appears that the test needs to be updated to reflect the changes in the annotator logic.
I suggest these changes:
```suggestion
                final HashProvider hash = new HashProviderFactory().getProvider(config.getHash().getType());
                List<String> hashesAndPaths = new ArrayList<>();
                hashesAndPaths.add(hash.derive(Files.readAllBytes(f1.toPath())).toLowerCase() + "  "
                                + f1.toPath().toString().replace(sourceCodeDir.toPath().toString(), "."));
                hashesAndPaths.add(hash.derive(Files.readAllBytes(f2.toPath())).toLowerCase() + "  "
                                + f2.toPath().toString().replace(sourceCodeDir.toPath().toString(), "."));
                Collections.sort(hashesAndPaths, Collator.getInstance(Locale.US));

                String hashesAndFiles = String.join("\n", hashesAndPaths) + "\n";
                final String checksum = hash.derive(hashesAndFiles.getBytes()).toLowerCase();
```
After implementing these changes, I have verified that all tests pass successfully. I have attached a screenshot for your reference.
![image](https://github.com/project-alvarium/alvarium-sdk-java/assets/29122581/0c1b4bbd-9580-402e-bc6f-a1ac49011f1b)

Reply:
Thanks for catching that, I must've been too caught up in testing on Jenkins that I forgot to run the unit tests hahaha. 

I added your suggestion in the latest commit, just lifted some of the logic into some additional variables.